### PR TITLE
Feature/748 750 parser and web resolver settings

### PR DESCRIPTION
**Pull request by marcovisserFurore**
Solves #748 and #750. 

This PR is part of PR #751 thanks to @wmrutten

### CODE REVIEW

File: src/Hl7.Fhir.Specification/Specification/Source/DirectorySource.cs [hunk 1]
```diff
@@ -884,32 +891,47 @@ private List<ArtifactSummary> harvestSummaries(List<string> paths)
...
+            var settings = _settings;
+            var factory = _navigatorFactory;
+            settings.XmlParserSettings.CopyTo(factory.XmlParsingSettings);
+            settings.JsonParserSettings.CopyTo(factory.JsonParsingSettings);
+
+            // Also use the current PoCo parser settings
+            var pocoSettings = new PocoBuilderSettings();
+            if (settings.ParserSettings != null)
             {
+                pocoSettings.AllowUnrecognizedEnums = settings.ParserSettings.AllowUnrecognizedEnums;
```
Comment (ewoutkramer) [inline src/Hl7.Fhir.Specification/Specification/Source/DirectorySource.cs]: Isn't this fragile?  What if we add new parser settings, these will then not get copied over?

File: src/Hl7.Fhir.Serialization/JsonNavigatorStream.cs [hunk 2]
```diff
@@ -55,14 +61,25 @@ public static JsonNavigatorStream FromPath(string path)
...
+        /// <summary>Create a new <see cref="JsonNavigatorStream"/> instance for the specified json resource stream.</summary>
+        /// <param name="stream">A stream that returns a serialized json resource.</param>
+        /// <param name="disposeStream">Determines if the <see cref="Dispose()"/> method should also dispose the specified <paramref name="stream"/> instance.</param>
+        /// <param name="settings">Parser settings that control de-serialization behavior.</param>
+        public JsonNavigatorStream(Stream stream, bool disposeStream, FhirJsonParsingSettings settings)
         {
             _stream = stream ?? throw Error.ArgumentNull(nameof(stream));
             _reader = SerializationUtil.JsonReaderFromStream(stream);
             _disposeStream = disposeStream;
+            ParserSettings = settings ?? FhirJsonParsingSettings.CreateDefault();
```
Comment (ewoutkramer) [inline src/Hl7.Fhir.Serialization/JsonNavigatorStream.cs]: Shouldn't this be `ParserSettings = settings.Clone()`?

File: src/Hl7.Fhir.Specification/Specification/Source/DirectorySource.cs (hunk 1 above)
Comment (wmrutten) [reply src/Hl7.Fhir.Specification/Specification/Source/DirectorySource.cs]: Marco suggests to implement members on settings classes themselves to copy internal state to another settings class. For example:

```
ParserSettings.CopyTo(PocoBuilderSettings settings) { /* ... */ }
PocoBuilderSettings.CopyTo(ParserSettings settings) { /* ... */ }
```

This is much better maintainable, esp. when adding new properties.

File: src/Hl7.Fhir.Serialization/JsonNavigatorStream.cs (hunk 2 above)
Comment (wmrutten) [reply src/Hl7.Fhir.Serialization/JsonNavigatorStream.cs]: Agreed, the API should always clone incoming settings.
And we should ensure that all the settings classes have similar behavior.

Comment (marcovisserFurore) [reply src/Hl7.Fhir.Serialization/JsonNavigatorStream.cs]: I've changed this behaviour to 
`ParserSettings = settings?.Clone() ?? FhirJsonParsingSettings.CreateDefault();`

File: src/Hl7.Fhir.Specification/Specification/Source/DirectorySource.cs (hunk 1 above)
Comment (marcovisserFurore) [reply src/Hl7.Fhir.Specification/Specification/Source/DirectorySource.cs]: I've created a new method for this:
`public void ParserSettings.CopyTo(PocoBuilderSettings settings)`

**Comment by wmrutten** [comment]
Great job. The new `CopyTo` methods on the settings classes are definitely an improvement. Way easier to maintain when we need to add new settings property.

**Pull request merged**